### PR TITLE
core/list: add remove method

### PR DESCRIPTION
**Pull request by lebrush**
this adds a method to remove elements from a list

Required by: #6158

### CODE REVIEW

File: core/include/list.h [hunk 1]
```diff
@@ -61,14 +64,37 @@ static inline void list_add(list_node_t *node, list_node_t *new_node) {
...
+static inline list_node_t *list_remove(list_node_t *list, list_node_t *node)
+{
+    while (list->next) {
+        if (list->next == node) {
+            list->next = node->next;
+            return node;
+        }
+        list = list->next;
+    }
+    return NULL;
```
Comment (miri64) [inline core/include/list.h]: No import of `<stddef.h>` in this file (yet ;-)), so `NULL` is not defined (found by Murdock)

**Comment by lebrush** [comment]
@miri64 I don't have time to write unittests for it now. I'll open an issue for that. 
I'll add the include quickly this afternoon :-)

**Comment by miri64** [comment]
KK.

**Comment by miri64** [comment]
Arghs, damn emb6 and its non-specific naming. Will provide a fix...

**Comment by miri64** [comment]
https://github.com/RIOT-OS/RIOT/pull/6225 was merged

**Comment by miri64** [comment]
(please rebase)

**Comment by miri64** [comment]
((and squash ;-))

**Comment by lebrush** [comment]
@miri64 rebased! if I squash then I touch irrelevant lines for the remove method in the same commit, I don't think it is a good idea...

**Comment by miri64** [comment]
kk

**Comment by lebrush** [comment]
ahhhrrrggg not again! nhdp has a list_remove... I'll check into within today :tada:

**Comment by miri64** [comment]
It's `oonf_api`, not `nhdp` ;-)

**Comment by lebrush** [comment]
Can someone remove the `NEEDS SQUASHING` to make Murdock happy, please :-)

```
Running './dist/tools/pr_check/pr_check.sh master' x
Command output:

	Pull request needs squashing according to its labels set on GitHub
```

**Comment by lebrush** [comment]
:tada: thx!

**Pull request merged**